tests/drivers/pwm/pwm_loopback/boards/s32k5xxcvb_s32k566_r52.overlay

```diff
@@ -0,0 +1,7 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "s32k5xxcvb_s32k566_m7.overlay"
```

Comment:
would it be possible to have a single overlay under `/soc` instead of `boards/`?

Reply:
We still need SoC s32k566_m7 and s32k566_r52 overlay. Additionally, the choice of PWM pins depends on the board's layout. I think using board overlay makes more sense (e.g if we required to support other k5 boards in the future)